crates/holochain/src/core/workflow/app_validation_workflow.rs

```diff
@@ -829,36 +774,23 @@ async fn run_validation_callback(
         ValidateResult::UnresolvedDependencies(UnresolvedDependencies::Hashes(hashes)) => {
             tracing::debug!(
                 ?hashes,
-                "Op validation returned unresolved dependencies -  AgentActivity"
+                "Op validation returned unresolved dependencies -  Hashes"
             );
             // fetch all missing hashes in the background without awaiting them
             let cascade_workspace = workspace.clone();
             let cascade =
                 CascadeImpl::from_workspace_and_network(&cascade_workspace, network.clone());
 
-            // keep track of which dependencies are being fetched for which dht op and to
-            // prevent multiple fetches of the same hash
-            let new_hashes_to_fetch = validation_dependencies
-                .lock()
-                .insert_missing_hashes_for_op(hashes.clone(), dht_op_hash.clone());
-
             // build a collection of futures to fetch the missing hashes
-            let fetches = new_hashes_to_fetch.into_iter().map(move |hash| {
+            let fetches = hashes.clone().into_iter().map(move |hash| {
                 let cascade = cascade.clone();
-                let validation_dependencies = validation_dependencies.clone();
                 async move {
                     let result = cascade
                         .fetch_record(hash.clone(), NetworkGetOptions::must_get_options())
                         .await;
                     if let Err(err) = result {
                         tracing::warn!("error fetching dependent hash {hash:?}: {err}");
                     }
-                    // Dependency has been fetched and added to the cache
-                    // or an error occurred along the way.
-                    // In case of an error the hash is still removed from
-                    // the collection so that it will be tried again to be
-                    // fetched in the subsequent workflow run.
-                    validation_dependencies.lock().remove_missing_hash(&hash);
```

Comment:
Okay this is where we would need to do more work on app validation. I follow how this is relevant to the conversation now - this is slow on a timeout and we could retry sooner